s3/src/main/scala/geotrellis/spark/io/s3/util/S3StreamBytes.scala

```diff
@@ -0,0 +1,63 @@
+package geotrellis.spark.io.s3.util
+
+import geotrellis.util.StreamBytes
+import geotrellis.spark.io.s3._
+
+import com.amazonaws.services.s3.model._
+
+/**
+ * This class extends [[StreamBytes]] by reading chunks out of a GeoTiff on the
+ * AWS S3 server.
+ *
+ * @param request: A [[GetObjectRequest]] of the desired GeoTiff.
+ * @param client: The [[AmazonS3Client]] that retrieves the data.
+ * @param chunkSize: An Int that specifies how many bytes should be read in at a time.
+ * @return A new instance of S3StreamBytes.
+ */
+class S3StreamBytes(
```

Comment:
Since it turned out this was a "verby" class, I recommend a name change to `S3ByteStreamer`. Likewise for the parent  `StreamBytes`.